pandas/tseries/tests/test_tslib.py

```diff
@@ -177,6 +177,29 @@ def test_constructor_invalid(self):
         with tm.assertRaisesRegexp(ValueError, 'Cannot convert Period'):
             Timestamp(Period('1000-01-01'))
 
+    def test_constructor_positional(self):
+        with tm.assertRaises(TypeError):
+            Timestamp(2000, 1)
+        with tm.assertRaises(ValueError):
+            Timestamp(2000, 0, 1)
+        with tm.assertRaises(ValueError):
+            Timestamp(2000, 13, 1)
+        with tm.assertRaises(ValueError):
+            Timestamp(2000, 1, 0)
+        with tm.assertRaises(ValueError):
+            Timestamp(2000, 1, 32)
+
```

Comment:
add some tests where ``None`` is passed for various places ``Timestamp(2000, 1, 1, None, None)``
``Timestamp(2000, 1, None, ...)``